python/pyspark/pandas/tests/test_series_string.py

```diff
@@ -259,10 +255,11 @@ def test_string_replace(self):
         def repl(m):
             return m.group(0)[::-1]
 
-        self.check_func(lambda x: x.str.replace(r"[a-z]+", repl))
+        regex_pat = re.compile(r"[a-z]+")
+        self.check_func(lambda x: x.str.replace(regex_pat, repl, regex=True))
         # compiled regex with flags
         regex_pat = re.compile(r"WHITESPACE", flags=re.IGNORECASE)
-        self.check_func(lambda x: x.str.replace(regex_pat, "---"))
```

Comment:
Actually, I think we should also make the behaviour the same with pandas 2.0.0 at pandas API on Spark too. I forgot how we concluded this. Right @itholic ? and should upgrade migration guide too?

Reply:
Yeah, as mentioned in https://github.com/apache/spark/pull/41824/files#r1250524246, it is advisable to wait for the Spark 4.0 release and address the behavior instead of just making changes to pass the tests. Just created a new ticket SPARK-44276 in order not to forget.

Reply:
Okay, let's get this reverted for now.